### PR TITLE
Add unit tests for Svc/FramingProtocol

### PR DESCRIPTION
**Pull request by bocchino**
This PR adds unit tests for testing the F Prime protocol implementation. It also adds a few missing checks for return status in the F Prime protocol implementation.

Unit test line coverage is 100% as measured by gcovr.

### CODE REVIEW

File: Svc/FramingProtocol/FprimeProtocol.cpp [hunk 1]
```diff
@@ -127,6 +129,7 @@ DeframingProtocol::DeframingStatus FprimeDeframing::deframe(Types::CircularBuffe
     FW_ASSERT(buffer.getSize() >= size);
     buffer.setSize(size);
     ring.peek(buffer.getData(), size, FpFrameHeader::SIZE);
```
Comment (arizvi786) [inline Svc/FramingProtocol/FprimeProtocol.cpp:131]: Do we need to assign "const Fw::SerializeStatus status = ring.peek(...)" in line 131 before asserting in line 132?

**Review by arizvi786 — COMMENTED** [review]
@bocchino looks good, just one comment, please take a look.

File: Svc/FramingProtocol/FprimeProtocol.cpp (hunk 1 above)
Comment (bocchino) [reply Svc/FramingProtocol/FprimeProtocol.cpp]: Yes, good catch. Fixed.

**Review by arizvi786 — APPROVED** [review]
looks good!

**Review by kevin-f-ortega — APPROVED** [review]
LGTM

**Review by kevin-f-ortega — APPROVED** [review]

**Review by LeStarch — APPROVED** [review]
Accepting existing reviews.

**Pull request merged**